packages/platforms/react-native/android/src/main/java/com/bugsnag/android/performance/NativeBugsnagPerformanceImpl.java

```diff
@@ -240,6 +253,16 @@ private SpanOptions readableMapToSpanOptions(ReadableMap jsOptions) {
     return spanOptions;
   }
 
+  private void discardLongRunningSpans() {
+    long oneHourAgo = SystemClock.elapsedRealtimeNanos() - TimeUnit.HOURS.toNanos(1);
+    for (Map.Entry<String, SpanImpl> entry : openSpans.entrySet()) {
+      SpanImpl span = entry.getValue();
+      if (span.getStartTime$internal() < oneHourAgo) {
+        openSpans.remove(entry.getKey());
+        span.discard();
+      }
+    }
```

Comment:
nit: I think an `Iterator.remove` is preferable to `Map.remove` here

```suggestion
    Iterator<Map.Entry<String, SpanImpl>> entries = openSpans.entrySet();
    while (entries.hasNext()) {
      Map.Entry<String, SpanImpl> entry = entries.next();
      SpanImpl span = entry.getValue();
      if (span.getStartTime$internal() < oneHourAgo) {
        entries.remove();
        span.discard();
      }
    }
```

Reply:
Updated 👍🏾